Projects/Claims/Sources/SubmitClaimStore.swift

```diff
@@ -246,21 +246,33 @@ public final class SubmitClaimStore: LoadingStateStore<SubmitClaimsState, Submit
             switch action {
             case let .setPhoneNumber(model):
                 newState.phoneNumberStep = model
-                send(.navigationAction(action: .openPhoneNumberScreen(model: model)))
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+                    self.send(.navigationAction(action: .openPhoneNumberScreen(model: model)))
+                }
             case let .setDateOfOccurrencePlusLocation(model):
                 newState.dateOfOccurrencePlusLocationStep = model.dateOfOccurencePlusLocationModel
                 newState.locationStep = model.locationModel
                 newState.dateOfOccurenceStep = model.dateOfOccurenceModel
-                send(.navigationAction(action: .openDateOfOccurrencePlusLocationScreen(options: [.date, .location])))
+                DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
```

Comment:
@sladan-hedvig I know this is not optimal but not sure if we can rework this in any way.